src/opml.cpp

```diff
@@ -107,55 +110,30 @@ void rec_find_rss_outlines(
 					xmlFree(filtercmd);
 				}
 
-				xmlFree(url);
 				// Filters and scripts may have arguments, so,
 				// quote them when needed.
-				// TODO: get rid of xmlStrdup, it's useless
-				url = (char*)xmlStrdup(
-						(const xmlChar*)
-						utils::quote_if_necessary(nurl)
-						.c_str());
-				assert(url);
-
-				bool found = false;
+				const std::string quoted_url = utils::quote_if_necessary(nurl);
 
 				LOG(Level::DEBUG,
 					"opml::import: size = %" PRIu64,
 					static_cast<uint64_t>(urlcfg.get_urls().size()));
-				// TODO: replace with algorithm::any or something
-				if (urlcfg.get_urls().size() > 0) {
-					for (const auto& u :
-						urlcfg.get_urls()) {
-						if (u == url) {
-							found = true;
-						}
-					}
-				}
 
-				if (!found) {
-					LOG(Level::DEBUG,
-						"opml::import: added url = %s",
-						url);
-					urlcfg.get_urls().push_back(
-						std::string(url));
+				auto& urls = urlcfg.get_urls();
+				if (std::find(urls.begin(), urls.end(), quoted_url) == urls.end()) {
+					LOG(Level::DEBUG, "opml::import: added url = %s", quoted_url);
+					urls.push_back(std::string(quoted_url));
```

Comment:
`quoted_url` is already of required type, so I think the call to `std::string` can be dropped here.

Reply:
Good point, done: https://github.com/newsboat/newsboat/pull/1310/commits/8968822269c33dac06e2a662f97035917693c4a9